lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object.rb

```diff
@@ -250,7 +250,7 @@ def process_assertions(message)
 
     def user_info_attributes(user)
       {'user' => user, 'tenant' => user.current_tenant, 'miq_group' => user.current_group}.each do |k, v|
-        value = MiqAeObject.convert_value_based_on_datatype(v.id, v.class.name)
+        value = MiqAeMethodService::MiqAeServiceModelBase.wrap_results(v)
```

Comment:
@mkanoor The ```find``` gets the user object from DB which is not what we want here since the user's current group from DB is not the correct requester group. 

We should use the in-memory user object that we already have at hand and wrap it into service model object. This way we don't need to change the expose of ```current_group``` in ```MiqAeServiceUser``` because the user object has got correct current group in memory. ```expose``` itself does not query the DB.